IPython/lib/deepreload.py

```diff
@@ -269,6 +271,12 @@ def deep_import_hook(name, globals=None, locals=None, fromlist=None, level=-1):
 
 def deep_reload_hook(m):
     """Replacement for reload()."""
+    # Hardcode this one  as it would raise a NotImplemeentedError from the
+    # bowels of Python and screw up the import machinery after.
+    # unlike other imports the `exclude` list aleady in place is not enough.
+
+    if m is types:
```

Comment:
Bit disturbing that the exclude list doesn't work for this.

I'd like to find a way to exclude all standard library modules from `deepreload()`, but I don't know of a good way to check if something's in the stdlib.

Maybe the rule should be 'anything that IPython uses' - i.e. anything that's in `sys.modules` before user code runs.  It's possibly tricky to do that now that we don't load `deepreload` on start.